tstest/integration/integration_test.go

```diff
@@ -872,6 +873,92 @@ func TestLogoutRemovesAllPeers(t *testing.T) {
 	wantNode0PeerCount(expectedPeers) // all existing peers and the new node
 }
 
+func TestAutoUpdateDefaults(t *testing.T) {
+	tstest.Shard(t)
+	tstest.Parallel(t)
+	env := newTestEnv(t)
+
+	sendAndCheckDefault := func(t *testing.T, n *testNode, send, want bool) {
+		t.Helper()
+		if !env.Control.AddRawMapResponse(n.MustStatus().Self.PublicKey, &tailcfg.MapResponse{
+			DefaultAutoUpdate: opt.NewBool(send),
+		}) {
+			t.Fatal("failed to send MapResponse to node")
+		}
+		if err := tstest.WaitFor(2*time.Second, func() error {
```

Comment:
Suggestion: I might give this function a meaningful name, and define it 1 or 2 levels of nesting up. This style is kind of nest-y. But, up to you.